### PR TITLE
Remove rpc error when shut down daemon

### LINKED ISSUE #22010: Don't show containerd connection closing as error
When the daemon shuts down, an error message is shown that is "expected" but probably redundant

To reproduce;

- manually start the docker daemon in the foreground; `docker daemon -D`
- after the daemon has finished starting, press `ctrl-c` to stop

This error is shown:

    ERRO[0117] failed to receive event from containerd: rpc error: code = 13 desc = "transport is closing"`

Docker version:

```
Client:
 Version:      1.11.0
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   4dc5990
 Built:        Wed Apr 13 18:13:28 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.0
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   4dc5990
 Built:        Wed Apr 13 18:13:28 2016
 OS/Arch:      linux/amd64
```

### PR DESCRIPTION
**Pull request by WeiZhang555**
Fixes #22010

RPC connection closing error will be reported every time we shutdown
daemon, this error is expected, so we should remove this error to avoid
confusion to user.

Signed-off-by: Zhang Wei zhangwei555@huawei.com

### CODE REVIEW

**Comment by thaJeztah** [comment]
Thanks! ping @tonistiigi @mlaventure for review :+1:

**Comment by thaJeztah** [comment]
tentatively adding this to 1.11.1 milestone, but it's just a minor improvement, so it's okay if it's not included

File: libcontainerd/remote_linux.go [hunk 1]
```diff
@@ -254,6 +256,11 @@ func (r *remote) handleEventStream(events containerd.API_EventsClient) {
 	for {
 		e, err := events.Recv()
 		if err != nil {
+			if strings.Contains(err.Error(), "transport is closing") &&
```
Comment (cpuguy83) [inline libcontainerd/remote_linux.go]: I was hoping to find a way to inspect this error without resorting to checking the error string, but it doesn't seem that this is possible with this error, which is a `grpc.rpcError`

Maybe we can do `strings.Contains(err.Error(), transport.ErrConnClosing.Desc)`

Comment (cpuguy83) [inline libcontainerd/remote_linux.go]: Actually looks like you can do `if grpc.ErrorDesc(err) == transport.ErrConClosing.Desc`... probably.

Comment (WeiZhang555) [inline libcontainerd/remote_linux.go]: Yes, I agree, inspect error string is stupid way, this looks better.
At first I thought it is `transport.ErrConClosing.Desc` but it's not, it's `rpcError` with Description from `transport.ErrConClosing`. I was hoping to find some function like `IsErrConClosing` but there's not something like this. :sob:

**Comment by WeiZhang555** [comment]
Updated.

**Comment by cpuguy83** [comment]
LGTM

**Comment by crosbymichael** [comment]
LGTM

**Pull request merged**